### DISCUSSION TITLE
How use WinMerge with GitHub Desktop?

### DISCUSSION THREAD

**Discussion by tinebak** [WinMerge/winmerge#1794] [Q&A] [answered]
I'm trying to use WinMerge with GitHub Drsktop to compare files between commit versions. I have added the following settings and restarted GitHib:

In .gitconfig

	[diff]
		tool = winmerge
	[difftool "winmerge"]
		name = WinMerge
		trustExitCode = true
		cmd = "/c/Program\\ Files\\/WinMerge/WinMergeU.exe" -u -e $LOCAL $REMOTE

GitHib currently shows 5 changed files in the Changes tab. I want to compare each of these files to the previous committed filed. (Or, sometimes, to an older commit).

I right-click the tab with the repository name and choose "Open in Command Prompt". This opens a command prompt. I then type in

    git difftool

This opens WinMerge. However, the only file that shows is the first file from the Changes tab.

How do I choose to compare other files? I've tried to right-click a specific file in GitHub, but I don't get an option to open in WinMerge or command prompt. I've tried various menu options in WinMerge. I've tried to enter file paths directly into File > Open, but I don't know the path to an old commit file version.

I have looked at other questions, including on Stackoverflow, but the answers I've found all leave off after entering the "git difftool" command.

**Comment by sdottaka** [comment, marked as answer]
What about `git difftool -d` instead of `git difftool`?

**Reply by tinebak** [reply]
Yes, thank you!! Now WinMerge opens with a directory tab of all files in the Changes tab. Do you know how to choose other versions to compare? For example,  files from two committed versions or current files with a commit version that is two commits back?

**Reply by sdottaka** [reply]
Specify two `<commit>`s as follows. I think the following URL will be helpful.
https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection
~~~
git difftool -d 2c40a4356d85f4527582ff6ceaf3374f24de1265 a5964c10e302f7c9430733bcfd28817be33bc08a
git difftool -d HEAD HEAD~
~~~

**Reply by tinebak** [reply]
Thank you. I'll play around with that. I saw that I can get the ID of the template by right clicking the commit version in the History pane and choosing Copy SHA. I was hoping for an actual GUI way of doing it, but now I know this way. I'll use the built-in github file compare split view for most times and this way when I need to see the full files next to each other.
